common/src/runners/mod.rs

```diff
@@ -188,11 +207,110 @@ impl BaseRunner {
+                result.push(TrustEntry::new(from.clone(), to.clone(), lt_entry_value));
+            }
+        } else {
+            let from = rev_domain_indices.get(&from_peer_start).unwrap();
+            let lt_row = lt
+                .get(&from_peer_start)
+                .ok_or(Error::OutboundLocalTrustNotFound(from_peer_start))?;
+            for to_peer_id in to_peer_start..lt_peers_cnt {
+                let to = rev_domain_indices.get(&to_peer_id).unwrap();
+                let lt_entry_value = lt_row.get(&to_peer_id).unwrap_or_default();
+                result.push(TrustEntry::new(from.clone(), to.clone(), lt_entry_value));
+            }
+
+            for from_peer_id in from_peer_start + 1..from_peer_end {
+                let from = rev_domain_indices.get(&from_peer_id).unwrap();
+                let lt_row =
+                    lt.get(&from_peer_id).ok_or(Error::OutboundLocalTrustNotFound(from_peer_id))?;
+                for to_peer_id in 0..lt_peers_cnt {
+                    let to = rev_domain_indices.get(&to_peer_id).unwrap();
+                    let lt_entry_value = lt_row.get(&to_peer_id).unwrap_or_default();
+                    result.push(TrustEntry::new(from.clone(), to.clone(), lt_entry_value));
+                }
+            }
+
+            if from_peer_end != lt_peers_cnt {
+                let from = rev_domain_indices.get(&from_peer_end).unwrap();
+                let lt_row = lt
+                    .get(&from_peer_end)
+                    .ok_or(Error::OutboundLocalTrustNotFound(from_peer_end))?;
+                for to_peer_id in 0..to_peer_end {
+                    let to = rev_domain_indices.get(&to_peer_id).unwrap();
+                    let lt_entry_value = lt_row.get(&to_peer_id).unwrap_or_default();
+                    result.push(TrustEntry::new(from.clone(), to.clone(), lt_entry_value));
+                }
+            }
+        }
+
+        let next_token = create_localtrust_next_token(lt_peers_cnt, from_peer_end, to_peer_end);
+
+        Ok(LocalTrustStateResponse::new(result, next_token))
+    }
+
+    pub fn get_st_state(
+        &self, domain: &Domain, page_size: Option<usize>, next_token: Option<String>,
+    ) -> Result<SeedTrustStateResponse, Error> {
+        let rev_domain_indices = self
+            .rev_indices
+            .get(&domain.to_hash())
+            .ok_or::<Error>(Error::ReverseIndicesNotFound(domain.to_hash()))?;
+        let st = self
+            .seed_trust
+            .get(&domain.seed_namespace())
+            .ok_or(Error::SeedTrustNotFound(domain.seed_namespace()))?;
+
+        let st_peers_cnt = st.len() as u64;
+        let (start_peer, end_peer) =
+            compute_seedtrust_peer_range(st_peers_cnt, page_size, next_token)?;
+
+        let mut result = vec![];
+        for peer_id in start_peer..end_peer {
+            let peer = rev_domain_indices.get(&peer_id).unwrap();
+            let seed = st.get(&peer_id).copied().unwrap_or_default();
```

Comment:
Seed entry should be skipped if does not exist

Reply:
Done https://github.com/openrankprotocol/openrank/pull/220/commits/6012248a917aa6e65b3bc8be0c81d3a8bf035cd2